bigquery_etl/shredder/config.py

```diff
@@ -110,9 +110,24 @@ def fields(self) -> Tuple[str, ...]:
         field="SUBSTR(hmac_user_id, 0, 32)",
     ),
 )
+LEGACY_MOBILE_SOURCES = tuple(
+    DeleteSource(
+        table=f"{product}_stable.deletion_request_v1",
+        field="metrics.uuid.legacy_ids_client_id",
+    )
+    for product in (
+        "org_mozilla_ios_fennec",
+        "org_mozilla_ios_firefox",
+        "org_mozilla_ios_firefoxbeta",
+        "org_mozilla_tv_firefox",
+        "mozilla_lockbox",
+    )
+)
 SOURCES = [DESKTOP_SRC, IMPRESSION_SRC, CFR_SRC, FXA_HMAC_SRC, FXA_SRC] + list(
     SYNC_SOURCES
-)
+) + list(LEGACY_MOBILE_SOURCES)
+
+LEGACY_MOBILE_IDS = tuple(CLIENT_ID for _ in LEGACY_MOBILE_SOURCES)
```

Comment:
This seems strange to me, why is a tuple of length `len(LEGACY_MOBILE_SOURCES)` full of the `client_id` literal string needed as a field in the `DeleteTarget` below?

Reply:
when specifying multiple sources, each source is allowed to match a different field in the target. `target.fields` and `sources` need to match in length for when they are `zip`'d together:

https://github.com/mozilla/bigquery-etl/blob/957e63c9a80449d79f0b404dd3464c3d477a84e6/bigquery_etl/shredder/delete.py#L188-L200

which results in DML like:
```
DELETE
  `moz-fx-data-shared-prod.telemetry_stable.core_v1`
WHERE
  (
    client_id IN (
      SELECT
        metrics.uuid.legacy_ids_client_id
      FROM
        `moz-fx-data-shared-prod.org_mozilla_ios_fennec_stable.deletion_request_v1`
      WHERE
        DATE(submission_timestamp) >= '2020-09-03'
        AND DATE(submission_timestamp) < '2020-09-17'
    )
    OR client_id IN (
      SELECT
        metrics.uuid.legacy_ids_client_id
      FROM
        `moz-fx-data-shared-prod.org_mozilla_ios_firefox_stable.deletion_request_v1`
      WHERE
        DATE(submission_timestamp) >= '2020-09-03'
        AND DATE(submission_timestamp) < '2020-09-17'
    )
    OR client_id IN (
      SELECT
        metrics.uuid.legacy_ids_client_id
      FROM
        `moz-fx-data-shared-prod.org_mozilla_ios_firefoxbeta_stable.deletion_request_v1`
      WHERE
        DATE(submission_timestamp) >= '2020-09-03'
        AND DATE(submission_timestamp) < '2020-09-17'
    )
    OR client_id IN (
      SELECT
        metrics.uuid.legacy_ids_client_id
      FROM
        `moz-fx-data-shared-prod.org_mozilla_tv_firefox_stable.deletion_request_v1`
      WHERE
        DATE(submission_timestamp) >= '2020-09-03'
        AND DATE(submission_timestamp) < '2020-09-17'
    )
    OR client_id IN (
      SELECT
        metrics.uuid.legacy_ids_client_id
      FROM
        `moz-fx-data-shared-prod.mozilla_lockbox_stable.deletion_request_v1`
      WHERE
        DATE(submission_timestamp) >= '2020-09-03'
        AND DATE(submission_timestamp) < '2020-09-17'
    )
  )
  AND CAST(submission_timestamp AS DATE) < '2020-09-17'
```